Mathlib/Logic/Relator.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2017 Johannes Hölzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl
+Relator for functions, pairs, sums, and lists.
+-/
+
+import Mathlib.Logic.Basic
+
+namespace relator
+universe u₁ u₂ v₁ v₂
+
+/- TODO(johoelzl):
+ * should we introduce relators of datatypes as recursive function or as inductive
+predicate? For now we stick to the recursor approach.
+ * relation lift for datatypes, Π, Σ, set, and subtype types
+ * proof composition and identity laws
+ * implement method to derive relators from datatype
+-/
+
+section
+
+variable {α : Sort u₁} {β : Sort u₂} {γ : Sort v₁} {δ : Sort v₂}
+variable (R : α → β → Prop) (S : γ → δ → Prop)
+
+/-- The binary relations `R : α → β → Prop` and `S : γ → δ → Prop` induce a binary
+    relation on functions `lift_fun : (f : α → γ) (g : β → δ) : Prop'. -/
+def lift_fun (f : α → γ) (g : β → δ) : Prop :=
```

Comment:
Mathport makes all of these defs CamelCase (e.g. `LiftFun`), but it's unclear to me whether we've settled on a convention for this kind of thing.

Reply:
After an offline discussion with @digama0, I've renamed all of these to match the output of synport.